### PR TITLE
Add Events Card/Extract VirtualList from events.

### PR DESCRIPTION
**Pull request by rawagner**
This PR replaces https://github.com/openshift/console/pull/1818

events

In this case Im using `Firehose` to fetch events as suggested by @spadgett . I still did some changes to existing `events.jsx` - I've extracted Event's `VirtualList` which is reused in Events dashboard card.

### CODE REVIEW

File: frontend/public/components/dashboard/events-card/events-card.scss [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  min-height: 10em;
+  overflow-x: hidden;
+  overflow-y: auto;
+}
+
+.co-events-card__body-stream {
+  margin-top: 1.5em;
+}
+
+.co-events-card__body-loading {
```
Comment (spadgett) [inline frontend/public/components/dashboard/events-card/events-card.scss]: nit: alphabetize declarations

File: frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { DashboardCard, DashboardCardHeader, DashboardCardBody, DashboardCardTitle } from '../../dashboard/dashboard-card';
+import { EventsBody } from '../../dashboard/events-card/events-body';
+import { DashboardItemProps, withDashboardResources } from '../with-dashboard-resources';
+import { EventModel } from '../../../models';
+import { FirehoseResult } from '../../utils';
+import { EventKind } from '../../../module/k8s';
+
+const eventsResource = {isList: true, kind: EventModel.kind, prop: 'events'};
```
Comment (spadgett) [inline frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx]: 👍

File: frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { DashboardCard, DashboardCardHeader, DashboardCardBody, DashboardCardTitle } from '../../dashboard/dashboard-card';
```
Comment (spadgett) [inline frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx]: Current convention is to separate external imports from console imports with a blank line

File: frontend/public/module/k8s/index.ts [hunk 4]
```diff
@@ -617,3 +619,17 @@ export type GroupVersionKind = string;
...
+  involvedObject: EventInvolvedObject;
+  message: string;
+  lastTimestamp: string;
+  firstTimestamp: string;
+  reason: string;
+  source: {
+    component: string;
+    host?: string;
+  }
+}
```
Comment (spadgett) [inline frontend/public/module/k8s/index.ts]:
```suggestion
};
```

File: frontend/public/module/k8s/event.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export type EventInvolvedObject = {
+  kind: string;
+  name: string;
+  uid: string;
+  namespace?: string;
+}
```
Comment (spadgett) [inline frontend/public/module/k8s/event.ts]:
```suggestion
};
```

Comment (rawagner) [reply frontend/public/module/k8s/event.ts]: added

File: frontend/public/module/k8s/index.ts (hunk 4 above)
Comment (rawagner) [reply frontend/public/module/k8s/index.ts]: added

File: frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx (hunk 3 above)
Comment (rawagner) [reply frontend/public/components/dashboards-page/overview-dashboard/events-card.tsx]: blank line added

File: frontend/public/components/dashboard/events-card/events-card.scss (hunk 1 above)
Comment (rawagner) [reply frontend/public/components/dashboard/events-card/events-card.scss]: fixed

**Comment by rawagner** [comment]
@spadgett thank you for the review. Fixed all.

**Pull request merged**